### ISSUE TITLE
KML with empty z coordinates results in features with null geometry

### ISSUE THREAD

**Issue by mike-000** [openlayers/openlayers#11714]
**Describe the bug**
ol/format/KML does not produce a geometry when placemark coordinates are in the format `<coordinates>6.28240681,49.50119400,</coordinates>`  A workaround is to load the file and replace each `,</coordinates>` by `</coordinates>` or `,0</coordinates>` but that should not be necessary as they are handled correctly by Google Earth.

Additionally ol/format/KML produces the default style instead of  #wronline when defined as 
```
<StyleMap id="wronline"><Pair><key>normal</key><styleUrl>n_wronline</styleUrl></Pair><Pair><key>highlight</key><styleUrl>h_wronline</styleUrl></Pair></StyleMap>
```
while a workaround to insert # 
```
<StyleMap id="wronline"><Pair><key>normal</key><styleUrl>#n_wronline</styleUrl></Pair><Pair><key>highlight</key><styleUrl>#h_wronline</styleUrl></Pair></StyleMap>
```
does work.

`**To Reproduce**
Use this example
```
<?xml version="1.0"?>
<kml><Document><name>WROnline</name><Style id="n_wronline"><IconStyle><scale>1.1</scale><hotSpot x="0.5" y="0" xunits="fraction" yunits="fraction"></hotSpot></IconStyle><LabelStyle><color></color><scale></scale></LabelStyle><LineStyle><color>011098008</color><width>3</width><gx:outerColor></gx:outerColor><gx:outerWidth></gx:outerWidth></LineStyle><PolyStyle><color>127193052</color><fill>1</fill><outline>1</outline></PolyStyle></Style><Style id="h_wronline"><IconStyle><scale>1.0</scale><hotSpot x="0.5" y="0" xunits="fraction" yunits="fraction"></hotSpot></IconStyle><LabelStyle><color></color><scale></scale></LabelStyle><LineStyle><color>234161014</color><width>3</width><gx:outerColor></gx:outerColor><gx:outerWidth></gx:outerWidth></LineStyle><PolyStyle><color>254224073</color><fill>1</fill><outline>1</outline></PolyStyle></Style><Style id="n_wronline_removed"><IconStyle><scale>1.1</scale><hotSpot x="0.5" y="0" xunits="fraction" yunits="fraction"></hotSpot></IconStyle><LabelStyle><color></color><scale></scale></LabelStyle><LineStyle><color>011098008</color><width>3</width><gx:outerColor></gx:outerColor><gx:outerWidth></gx:outerWidth></LineStyle><PolyStyle><color>127193052</color><fill>1</fill><outline>1</outline></PolyStyle></Style><Style id="h_wronline_removed"><IconStyle><scale>1.0</scale><hotSpot x="0.5" y="0" xunits="fraction" yunits="fraction"></hotSpot></IconStyle><LabelStyle><color></color><scale></scale></LabelStyle><LineStyle><color>234161014</color><width>3</width><gx:outerColor></gx:outerColor><gx:outerWidth></gx:outerWidth></LineStyle><PolyStyle><color>254224073</color><fill>1</fill><outline>1</outline></PolyStyle></Style><StyleMap id="wronline"><Pair><key>normal</key><styleUrl>n_wronline</styleUrl></Pair><Pair><key>highlight</key><styleUrl>h_wronline</styleUrl></Pair></StyleMap><StyleMap id="wronline_removed"><Pair><key>normal</key><styleUrl>n_wronline_removed</styleUrl></Pair><Pair><key>highlight</key><styleUrl>h_wronline_removed</styleUrl></Pair></StyleMap><Folder><name>Luxembourg</name><Placemark><name>&lt;a href="https://www.spottingmode.com/wro/location/10098/"&gt;Luxembourg&lt;/a&gt;</name><description>AA-5</description><styleUrl>#wronline</styleUrl><Point><coordinates>6.19474459,49.62422943,</coordinates></Point></Placemark><Placemark><name>&lt;a href="https://www.spottingmode.com/wro/location/19372/"&gt;Luxembourg&lt;/a&gt;</name><description>MD.902</description><styleUrl>#wronline</styleUrl><Point><coordinates>6.19928074,49.62609482,</coordinates></Point></Placemark><Placemark><name>&lt;a href="https://www.spottingmode.com/wro/location/17046/"&gt;Mondorf-les-Bains - Musee de l'Aviation Luxembourgeoise&lt;/a&gt;</name><description>Various</description><styleUrl>#wronline</styleUrl><Point><coordinates>6.28240681,49.50119400,</coordinates></Point></Placemark><Placemark><name>&lt;a href="https://www.spottingmode.com/wro/location/22995/"&gt;Wemperhardt&lt;/a&gt;</name><description>Various</description><styleUrl>#wronline</styleUrl><Point><coordinates>6.05678320,50.14853668,</coordinates></Point></Placemark></Folder></Document></kml>
```
or download any KML from https://www.spottingmode.com/wro/download/  If used in conjunction with the drag and drop example features are added to the source but attempting to fit to their extent fails as there is no geometry.

**Expected behavior**
The x and y coordinates should be used as in Google Earth and the empty z coordinate defaulted to 0.
Style maps defined without using # should work as they do in Google Earth.

**Comment by ahocevar** [comment]
Fixed with #11721

**Comment by mike-000** [comment]
Remaining problem fixed by #11732
